src/hooks/useCalculator.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import { isOperator, getCurrentDigits, OperatorType, caculationResult, OperationType } from '../utils';
+
+export const useCalculator = () => {
+  const [currentNumber, setCurrentNumber] = useState<number>(0);
+  const [operator, setOperator] = useState<OperatorType | null>(null);
+  const [result, setResult] = useState<string>('0');
+  const [firstNumber, setFirstNumber] = useState<number>(0);
+
+  // 1.  들어오는 Digit에따라 숫자를 설정하고  result컴포넌트에 결과를 보여준다 .
+  const insertDigit = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const currentDigitNumber = getCurrentDigits(currentNumber, e.currentTarget.value);
+    try {
+      if (currentDigitNumber.length > 3) {
+        throw new Error('최대 3자리까지 입력가능합니다.');
+      }
```

Comment:
```suggestion
    try {
       const currentDigitNumber = getCurrentDigits(currentNumber, e.currentTarget.value);
```
이렇게하고.. getCurrentDigits함수안에서 validation을하고 문제가 있다면 throw Error해주면 깔끔할 것 같아요

Reply:
고생하셨습니다 !  많이배웠습니다!